src/ConfigParser.cpp

```diff
@@ -0,0 +1,246 @@
+	ServerBlock server_block;
+	std::string line;
+	while (std::getline(stream, line))
+	{
+		line = stripInlineComments(line);
+		if (isCommentOrEmpty(line))
+			continue;
+		std::istringstream line_stream(line);
+		std::string first_word;
+		line_stream >> first_word;
+		if (first_word == "}")
+			break;
+		else if (first_word == "port")
+		{
+			std::string port;
+			line_stream >> port;
+			server_block.settings[ServerSetting::Port] = port;
+			logger.log(LogLevel::DEBUG, "Port: " + port);
+		}
+		else if (first_word == "host")
+		{
+			std::string host;
+			line_stream >> host;
+			server_block.settings[ServerSetting::Host] = host;
+			logger.log(LogLevel::DEBUG, "Host: " + host);
+		}
+		else if (first_word == "server_name")
+		{
+			std::string server_name;
+			line_stream >> server_name;
+			server_block.settings[ServerSetting::ServerName] = server_name;
+			logger.log(LogLevel::DEBUG, "ServerName: " + server_name);
+		}
+		else if (first_word == "client_max_body_size")
+		{
+			std::string client_max_body_size;
+			line_stream >> client_max_body_size;
+			server_block.settings[ServerSetting::ClientMaxBodySize] =
+				client_max_body_size;
+			logger.log(LogLevel::DEBUG,
+					   "ClientMaxBodySize: " + client_max_body_size);
+		}
+		else if (first_word == "error_pages")
+		{
+			std::string error_pages;
+			line_stream >> error_pages;
+			server_block.settings[ServerSetting::ErrorPages] = error_pages;
+			logger.log(LogLevel::DEBUG, "ErrorPages: " + error_pages);
+		}
+		else if (first_word == "location")
+		{
+			std::string prefix;
+			line_stream >> prefix;
+			parseLocationBlock(stream, prefix);
+		}
+		else
+			logger.log(LogLevel::WARNING,
+					   "Unkown location setting: " + first_word);
+	}
+	_server_blocks.push_back(server_block);
+	logger.log(LogLevel::DEBUG, "Added server block\n");
+}
+
+void ConfigParser::parseGlobalBlock(std::istream &stream)
+{
+	Logger &logger = Logger::getInstance();
+	logger.log(LogLevel::DEBUG, "Parsing global block");
+	std::string line;
+	while (std::getline(stream, line))
+	{
+		line = stripInlineComments(line);
+		if (isCommentOrEmpty(line))
+			continue;
+		std::istringstream line_stream(line);
+		std::string first_word;
+		line_stream >> first_word;
+		if (first_word == "}")
+			break;
+		if (first_word == "threads")
+		{
+			std::string threads;
+			line_stream >> threads;
+			_global_settings[GlobalSetting::Threads] = threads;
+			logger.log(LogLevel::DEBUG, "Threads: " + threads);
+		}
+		else if (first_word == "default_error_pages")
+		{
+			std::string default_error_pages;
+			line_stream >> default_error_pages;
+			_global_settings[GlobalSetting::DefaultErrorPages] =
+				default_error_pages;
+			logger.log(LogLevel::DEBUG, "log_level: " + default_error_pages);
+		}
+		else
+			logger.log(LogLevel::WARNING,
+					   "Unkown global setting: " + first_word);
+	}
+	logger.log(LogLevel::DEBUG, "Added global block\n");
+}
+
+bool ConfigParser::isCommentOrEmpty(const std::string &line)
```

Comment:
I think it would be nice to have the ability to put comments at the end of a line aswell